src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,29 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
+import { setQuery, setStatus } from '../../features/filter';
+import { Status } from '../../types/Status';
+import { useTypedSelector } from '../../app/hooks';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+  const query = useTypedSelector(state => state.filter.query);
+
+  const handleChangeQuery = (e: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(setQuery(e.target.value));
+  };
+
+  const handleStatusChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(setStatus(e.target.value as Status));
```

Comment:
Ensure that the `Status` type is correctly defined and imported. This type casting is crucial for maintaining type safety when updating the filter status.